File: code/dynamic_programming/src/minimum_skips_to_arrive_at_meeting_on_time/Minimum Skips to Arrive at Meeting On Time.py
def minSkips(self, A, s, target):
     n = len(A)
      dp = [0] * (n + 1)
       for i, a in enumerate(A):
            for j in xrange(n, -1, -1):
                dp[j] += a
                if i < n - 1:
                    dp[j] = (dp[j] + s - 1) / s * s
                if j:
                    dp[j] = min(dp[j], dp[j - 1] + a)
        for i in xrange(n):
            if dp[i] <= s * target:
                return i
        return -1

File: code/dynamic_programming/src/minimum_skips_to_arrive_at_meeting_on_time/Minimum Skips to Arrive at Meeting On Time.java
public int minSkips(int[] A, int s, int target) {
        int n = A.length, dp[] = new int[n + 1];
        for (int i = 0; i < n; ++i) {
            for (int j = n; j >= 0; --j) {
                dp[j] += A[i];
                if (i < n - 1)
                    dp[j] = (dp[j] + s - 1) / s * s; // take a rest
                if (j > 0)
                    dp[j] = Math.min(dp[j], dp[j - 1] + A[i]);
            }
        }
        for (int i = 0; i < n; ++i) {
            if (dp[i] <= (long)s * target)
                return i;
        }
        return -1;
    }